File: src/leetcode/RemoveDuplicates.java
package leetcode;

public class RemoveDuplicates {

    public static void main(String[] args) {
        var test1 = removeDuplicates(new int[]{1, 1, 1, 2, 2, 3});
        System.out.println(test1);
    }

    public static int removeDuplicates(int[] nums) {
        var count = 1;
        var k = 1;

        for (int i = 1; i < nums.length; i++) {
            if (nums[i] == nums[i - 1]) {
                count++;
            } else {
                count = 1;
            }

            if (count <= 2) {
                nums[k] = nums[i];
                k++;
            }
        }
        return k;
    }

}

File: src/leetcode/kotlin/RemoveDuplicates.kt
package leetcode.kotlin

class RemoveDuplicates {

    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
//            println(removeDuplicates(intArrayOf(1, 1, 2)))
            println(removeDuplicates(intArrayOf(0, 0, 1, 1, 1, 2, 2, 3, 3, 4)))
        }

        fun removeDuplicates(nums: IntArray): Int {
            var k = 1
            var count = 1

            for (i in 1 until nums.size) {
                if (nums[i] == nums[i - 1]) {
                    count++
                } else {
                    count = 1
                }

                if (count <= 1) {
                    nums[k] = nums[i]
                    k++
                }
            }
            return k;
        }
    }
}